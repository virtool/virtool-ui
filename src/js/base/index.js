export { Accordion } from "./accordion/Accordion";
export { AccordionContent } from "./accordion/AccordionContent";
export { AccordionTrigger } from "./accordion/AccordionTrigger";
export { ScrollingAccordionItem } from "./accordion/ScrollingAccordionItem";
export { Alert } from "./Alert";
export { AlertInner } from "./AlertInner";
export { AlertOuter } from "./AlertOuter";
export { Attribution } from "./Attribution";
export { AttributionWithName } from "./AttributionWithName";
export { Badge } from "./Badge";
export { Box } from "./Box";
export { BoxGroup } from "./BoxGroup";
export { BoxGroupDisabled } from "./BoxGroupDisabled";
export { BoxGroupHeader } from "./BoxGroupHeader";
export { BoxGroupSearch } from "./BoxGroupSearch";
export { BoxGroupSection } from "./BoxGroupSection";
export { BoxGroupTable } from "./BoxGroupTable";
export { BoxSpaced } from "./BoxSpaced";
export { BoxTitle } from "./BoxTitle";
export { device } from "./Breakpoints";
export { Button } from "./Button";
export { Checkbox } from "./Checkbox";
export { CheckboxLabel } from "./CheckboxLabel";
export { Circle } from "./Circle";
export { CloseButton } from "./CloseButton";
export { Color } from "./Color";
export { ComboBox } from "./combobox/ComboBox";
export { Container } from "./Container";
export { ContainerNarrow } from "./ContainerNarrow";
export { ContainerSide } from "./ContainerSide";
export { ContainerWide } from "./ContainerWide";
export { Dialog } from "./Dialog";
export { DialogAlert } from "./DialogAlert";
export { DialogContent } from "./DialogContent";
export { DialogContext } from "./DialogContext";
export { DialogFooter } from "./DialogFooter";
export { DialogOverlay } from "./DialogOverlay";
export { DialogTabs } from "./DialogTabs";
export { DialogTitle } from "./DialogTitle";
export { DialogTrigger } from "./DialogTrigger";
export { DividerHorizontal } from "./DividerHorizontal";
export { DividerVertical } from "./DividerVertical";
export { Dropdown } from "./Dropdown";
export { DropdownButton } from "./DropdownButton";
export { DropdownMenuContent } from "./DropdownMenuContent";
export { DropdownMenuDownload } from "./DropdownMenuDownload";
export { DropdownMenuItem } from "./DropdownMenuItem";
export { DropdownMenuLink } from "./DropdownMenuLink";
export { ExternalLink } from "./ExternalLink";
export { Icon } from "./Icon";
export { IconButton } from "./IconButton";
export { InitialIcon } from "./InitialIcon";
export { Input } from "./Input";
export { InputContainer } from "./InputContainer";
export { InputContext } from "./InputContext";
export { InputError } from "./InputError";
export { InputGroup } from "./InputGroup";
export { InputIcon } from "./InputIcon";
export { InputIconButton } from "./InputIconButton";
export { InputLabel } from "./InputLabel";
export { InputLevel } from "./InputLevel";
export { InputLoading } from "./InputLoading";
export { InputPassword } from "./InputPassword";
export { InputSearch } from "./InputSearch";
export { InputSelect } from "./InputSelect";
export { InputSimple } from "./InputSimple";
export { Key } from "./Key";
export { Label } from "./Label";
export { LinkButton } from "./LinkButton";
export { Loader } from "./Loader";
export { LoadingPlaceholder } from "./LoadingPlaceholder";
export { Logo } from "./Logo";
export { Markdown } from "./Markdown";
export { NoneFound } from "./NoneFound";
export { NoneFoundBox } from "./NoneFoundBox";
export { NoneFoundSection } from "./NoneFoundSection";
export { NotFound } from "./NotFound";
export { Pagination } from "./pagination/Pagination";
export { ProgressBar } from "./ProgressBar";
export { ProgressBarAffixed } from "./ProgressBarAffixed";
export { ProgressCircle } from "./ProgressCircle";
export { RelativeTime } from "./RelativeTime";
export { RemoveBanner } from "./RemoveBanner";
export { SaveButton } from "./SaveButton";
export { SectionHeader } from "./SectionHeader";
export { Select } from "./select/Select";
export { SelectButton } from "./select/SelectButton";
export { SelectContent } from "./select/SelectContent";
export { SelectItem } from "./select/SelectItem";
export { SelectBox } from "./SelectBox";
export { SelectBoxGroupSection } from "./SelectBoxGroupSection";
export { SidebarHeader } from "./SidebarHeader";
export { SidebarHeaderButton } from "./SidebarHeaderButton";
export { SideBarSection } from "./SideBarSection";
export { SlashList } from "./SlashList";
export { SubviewHeader } from "./SubviewHeader";
export { SubviewHeaderAttribution } from "./SubviewHeaderAttribution";
export { SubviewHeaderTitle } from "./SubviewHeaderTitle";
export { Table } from "./Table";
export { Tabs } from "./Tabs";
export { TabsLink } from "./TabsLink";
export { TextArea } from "./TextArea";
export { Toolbar } from "./Toolbar";
export { ViewHeader } from "./ViewHeader";
export { ViewHeaderAttribution } from "./ViewHeaderAttribution";
export { ViewHeaderIcons } from "./ViewHeaderIcons";
export { ViewHeaderSubtitle } from "./ViewHeaderSubtitle";
export { ViewHeaderTitle } from "./ViewHeaderTitle";
