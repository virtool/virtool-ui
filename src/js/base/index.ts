import InputGroup from "@base/InputGroup.js";
import InputLabel from "@base/InputLabel.js";

export { InputGroup, InputLabel };

export { Link } from "wouter";
export { Checkbox } from "./Checkbox";
export { CloseButton } from "./CloseButton";
export { Color } from "./Color";
export { ComboBox } from "./ComboBox";
export { Container } from "./Container";
export { ContainerNarrow } from "./ContainerNarrow";
export { ContainerSide } from "./ContainerSide";
export { ContainerWide } from "./ContainerWide";
export { Dialog } from "./Dialog";
export { DialogAlert } from "./DialogAlert";
export { DialogContent } from "./DialogContent";
export { DialogContext } from "./DialogContext";
export { DialogFooter } from "./DialogFooter";
export { DialogOverlay } from "./DialogOverlay";
export { DialogTabs } from "./DialogTabs";
export { DialogTitle } from "./DialogTitle";
export { DialogTrigger } from "./DialogTrigger";
export { Dropdown } from "./Dropdown";
export { DropdownButton } from "./DropdownButton";
export { DropdownMenuContent } from "./DropdownMenuContent";
export { DropdownMenuDownload } from "./DropdownMenuDownload";
export { DropdownMenuItem } from "./DropdownMenuItem";
export { DropdownMenuLink } from "./DropdownMenuLink";
export { ExternalLink } from "./ExternalLink";
export { Icon } from "./Icon";
export { IconButton } from "./IconButton";
export { InitialIcon } from "./InitialIcon";
export { Input } from "./Input";
export { InputContainer } from "./InputContainer";
export { InputContext } from "./InputContext";
export { InputError } from "./InputError";
export { InputIcon } from "./InputIcon";
export { InputIconButton } from "./InputIconButton";
export { InputLoading } from "./InputLoading";
export { InputPassword } from "./InputPassword";
export { InputSearch } from "./InputSearch";
export { InputSelect } from "./InputSelect";
export { InputSimple } from "./InputSimple";
export { Key } from "./Key";
export { Label } from "./Label";
export { LinkButton } from "./LinkButton";
export { Loader } from "./Loader";
export { LoadingPlaceholder } from "./LoadingPlaceholder";
export { Logo } from "./Logo";
export { Markdown } from "./Markdown";
export { NoneFound } from "./NoneFound";
export { NoneFoundBox } from "./NoneFoundBox";
export { NoneFoundSection } from "./NoneFoundSection";
export { NotFound } from "./NotFound";
export { Pagination } from "./Pagination";
export { ProgressBar } from "./ProgressBar";
export { ProgressBarAffixed } from "./ProgressBarAffixed";
export { ProgressCircle } from "./ProgressCircle";
export { RelativeTime } from "./RelativeTime";
export { RemoveBanner } from "./RemoveBanner";
export { SaveButton } from "./SaveButton";
export { SectionHeader } from "./SectionHeader";
export { Select } from "./Select";
export { SelectBox } from "./SelectBox";
export { SelectBoxGroupSection } from "./SelectBoxGroupSection";
export { SelectButton } from "./SelectButton";
export { SelectContent } from "./SelectContent";
export { SelectItem } from "./SelectItem";
export { SidebarHeader } from "./SidebarHeader";
export { SidebarHeaderButton } from "./SidebarHeaderButton";
export { SideBarSection } from "./SideBarSection";
export { SlashList } from "./SlashList";
export { SubviewHeader } from "./SubviewHeader";
export { SubviewHeaderAttribution } from "./SubviewHeaderAttribution";
export { SubviewHeaderTitle } from "./SubviewHeaderTitle";
export { Table } from "./Table";
export { Tabs } from "./Tabs";
export { TabsLink } from "./TabsLink";
export { TextArea } from "./TextArea";
export { ViewHeader } from "./ViewHeader";
export { ViewHeaderAttribution } from "./ViewHeaderAttribution";
export { ViewHeaderIcons } from "./ViewHeaderIcons";
export { ViewHeaderSubtitle } from "./ViewHeaderSubtitle";
export { ViewHeaderTitle } from "./ViewHeaderTitle";
