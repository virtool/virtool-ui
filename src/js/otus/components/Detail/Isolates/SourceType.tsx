import { InputGroup, InputLabel, InputSelect, InputSimple } from "@base/index";
import { capitalize, map } from "lodash-es";
import React from "react";
import { UseFormRegister, UseFormWatch } from "react-hook-form";

type IsolateFormValues = {
    sourceName: string;
    sourceType: string;
};

type SourceTypeProps = {
    allowedSourceTypes: string[];
    /** Function to register form fields */
    register: UseFormRegister<IsolateFormValues>;
    /** Indicates whether the source types are restricted */
    restrictSourceTypes: boolean;
    /** Watches for changes in form values */
    watch: UseFormWatch<IsolateFormValues>;
};

/**
 * Displays input for source type in isolate creation dialog
 */
export function SourceType({
    allowedSourceTypes,
    register,
    restrictSourceTypes,
    watch,
}: SourceTypeProps) {
    if (restrictSourceTypes) {
        const optionComponents = map(allowedSourceTypes, (sourceType) => (
            <option key={sourceType} value={capitalize(sourceType)}>
                {capitalize(sourceType)}
            </option>
        ));

        return (
            <InputGroup>
                <InputLabel htmlFor="sourceType">Source Type</InputLabel>
                <InputSelect id="sourceType" {...register("sourceType")}>
                    <option key="default" value="unknown">
                        Unknown
                    </option>
                    {optionComponents}
                </InputSelect>
            </InputGroup>
        );
    }

    return (
        <InputGroup>
            <InputLabel htmlFor="sourceType">Source Type</InputLabel>
            <InputSimple
                id="sourceType"
                {...register("sourceType")}
                value={capitalize(watch("sourceType"))}
            />
        </InputGroup>
    );
}
