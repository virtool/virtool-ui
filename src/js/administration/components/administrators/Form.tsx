import React from "react";
import { Controller, useForm } from "react-hook-form";
import styled from "styled-components";
import { InputError, InputGroup, InputLabel, SaveButton } from "../../../base";
import { User } from "../../../users/types";
import { useFindUsers, useGetAdministratorRoles, useSetAdministratorRole } from "../../queries";
import { AdministratorRoles } from "../../types";
import { RoleSelect } from "./RoleSelect";
import { UserSelect } from "./UserSelect";

const RoleSelectContainer = styled.div`
    display: flex;
    flex-direction: column;

    button {
        flex-grow: 1;
        padding: 10px 10px;
    }
`;

const UserSelectContainer = styled.div`
    display: flex;
    flex-direction: column;
    button {
        flex-grow: 1;
        padding: 10px 10px;
    }
`;

interface FormInputValues {
    user: User;
    role: AdministratorRoles;
}

type AdministratorFormProps = {
    onClose: () => void;
};

export function AdministratorForm({ onClose }: AdministratorFormProps): JSX.Element {
    const {
        formState: { errors },
        handleSubmit,
        control,
    } = useForm<FormInputValues>();

    const [userSearchTerm, setUserSearchTerm] = React.useState("");

    const { data: users } = useFindUsers(1, 25, userSearchTerm, false);

    const { data: roles } = useGetAdministratorRoles();

    const administratorRoleMutator = useSetAdministratorRole();

    const onSubmit = ({ user, role }: FormInputValues) => {
        administratorRoleMutator.mutate({ user_id: user.id, role });
        onClose();
    };

    return (
        <form onSubmit={handleSubmit(onSubmit)}>
            <InputGroup>
                <InputLabel htmlFor="user">User</InputLabel>
                <UserSelectContainer>
                    <Controller
                        render={({ field: { onChange, value } }) => (
                            <UserSelect
                                onChange={onChange}
                                value={value}
                                users={users?.items || []}
                                onTermChange={setUserSearchTerm}
                                term={userSearchTerm}
                                id="user"
                            />
                        )}
                        name="user"
                        control={control}
                        rules={{ required: "A user must be selected" }}
                    />
                </UserSelectContainer>
                <InputError>{errors.user?.message}</InputError>
            </InputGroup>
            <InputGroup>
                <InputLabel htmlFor="role">Role</InputLabel>
                <RoleSelectContainer>
                    <Controller
                        render={({ field: { onChange, value } }) => (
                            <RoleSelect onChange={onChange} value={value} roles={roles || []} id="role" />
                        )}
                        name="role"
                        control={control}
                        rules={{ required: "A role must be selected" }}
                    />
                    <InputError>{errors.role?.message}</InputError>
                </RoleSelectContainer>
            </InputGroup>
            <SaveButton />
        </form>
    );
}
