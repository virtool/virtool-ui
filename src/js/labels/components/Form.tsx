import React, { useState } from "react";
import { useForm } from "react-hook-form";
import styled from "styled-components";
import { getFontWeight } from "../../app/theme";
import { Box, Button, Color, InputError, InputGroup, InputLabel, InputSimple } from "../../base";
import { SampleLabel } from "../../samples/components/Label";

const LabelFormPreview = styled(Box)`
    padding: 10px;
`;

const LabelFormPreviewLabel = styled.p`
    font-weight: ${getFontWeight("thick")};
`;

interface LabelFormProps {
    color?: string;
    description?: string;
    error?: string;
    name?: string;
    onSubmit: (data: { color: string; name: string; description: string }) => void;
}

export function LabelForm({ color = "#D1D5DB", description = "", error = "", name = "", onSubmit }: LabelFormProps) {
    const [newColor, setColor] = useState(color);

    const {
        formState: { errors },
        register,
        handleSubmit,
        watch
    } = useForm({ defaultValues: { color, description, name } });

    return (
        <form onSubmit={handleSubmit(values => onSubmit({ ...values, color: newColor || "#D1D5DB" }))}>
            <InputGroup>
                <InputLabel htmlFor="name">Name</InputLabel>
                <InputSimple
                    id="name"
                    aria-invalid={errors.name ? "true" : "false"}
                    {...register("name", { required: "Name is required." })}
                />
                <InputError>{errors.name?.message || error}</InputError>
            </InputGroup>
            <InputGroup>
                <InputLabel htmlFor="description">Description</InputLabel>
                <InputSimple
                    id="description"
                    aria-invalid={errors.description ? "true" : "false"}
                    {...register("description")}
                />
            </InputGroup>
            <InputGroup>
                <InputLabel htmlFor="color">Color</InputLabel>
                <Color id="color" value={newColor} onChange={color => setColor(color)} />
            </InputGroup>
            <LabelFormPreviewLabel>Preview</LabelFormPreviewLabel>
            <LabelFormPreview>
                <SampleLabel color={newColor || "#D1D5DB"} name={watch("name") || "Preview"} />
            </LabelFormPreview>
            <Button type="submit" color="blue" icon="save">
                Save
            </Button>
        </form>
    );
}
