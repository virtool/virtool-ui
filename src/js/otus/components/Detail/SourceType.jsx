import { capitalize, map } from "lodash-es";
import React from "react";
import { Input, InputGroup, InputLabel, InputSelect } from "../../../base";

export const SourceType = ({ restrictSourceTypes, allowedSourceTypes, value, onChange }) => {
    if (restrictSourceTypes) {
        const optionComponents = map(allowedSourceTypes, sourceType => (
            <option key={sourceType} value={capitalize(sourceType)}>
                {capitalize(sourceType)}
            </option>
        ));

        return (
            <InputGroup>
                <InputLabel>Source Type</InputLabel>
                <InputSelect value={capitalize(value)} onChange={onChange}>
                    <option key="default" value="unknown">
                        Unknown
                    </option>
                    {optionComponents}
                </InputSelect>
            </InputGroup>
        );
    }

    return (
        <InputGroup>
            <InputLabel>Source type</InputLabel>
            <Input value={capitalize(value)} onChange={onChange} />
        </InputGroup>
    );
};
