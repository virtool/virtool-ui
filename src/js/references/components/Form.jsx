import { Field } from "formik";
import React from "react";
import styled from "styled-components";
import { Input, InputError, InputGroup, InputLabel, TextArea } from "../../base";

const StyledInputGroup = styled(InputGroup)`
    padding-bottom: 0px;
`;

export const ReferenceForm = ({ errors, touched, mode }) => {
    let organismComponent;

    if (mode === "empty" || mode === "edit") {
        organismComponent = (
            <InputGroup>
                <InputLabel htmlFor="organism">Organism</InputLabel>
                <Field as={Input} name="organism" id="organism" />
            </InputGroup>
        );
    }
    const nameError = touched.name && errors.name;

    return (
        <>
            <StyledInputGroup>
                <InputLabel htmlFor="name">Name</InputLabel>
                <Field name="name" id="name" as={Input} error={nameError} />
                <InputError>{nameError}</InputError>
            </StyledInputGroup>

            {organismComponent}

            <InputGroup>
                <InputLabel htmlFor="description">Description</InputLabel>
                <Field as={TextArea} name="description" id="description" />
            </InputGroup>
        </>
    );
};
