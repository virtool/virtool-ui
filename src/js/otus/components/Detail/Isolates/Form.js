import { toLower } from "lodash-es";
import PropTypes from "prop-types";
import React from "react";
import styled from "styled-components";
import { ModalBody, ModalFooter, Input, InputGroup, InputLabel, SaveButton } from "../../../../base";

import { formatIsolateName } from "../../../../utils/utils";
import { SourceType } from "../SourceType";

const IsolateFormFields = styled.div`
    display: grid;
    grid-template-columns: 1fr 1fr;
    grid-column-gap: ${props => props.theme.gap.column};
`;

export class IsolateForm extends React.Component {
    static propTypes = {
        sourceType: PropTypes.string,
        sourceName: PropTypes.string,
        allowedSourceTypes: PropTypes.arrayOf(PropTypes.string),
        restrictSourceTypes: PropTypes.bool,
        onChange: PropTypes.func,
        onSubmit: PropTypes.func
    };

    handleChangeSourceType = e => {
        this.props.onChange({
            sourceType: toLower(e.target.value),
            sourceName: e.target.value === "unknown" ? "" : this.props.sourceName
        });
    };

    handleChangeSourceName = e => {
        this.props.onChange({
            sourceName: e.target.value,
            sourceType: this.props.sourceType
        });
    };

    render() {
        return (
            <form onSubmit={this.props.onSubmit}>
                <ModalBody>
                    <IsolateFormFields>
                        <SourceType
                            restrictSourceTypes={this.props.restrictSourceTypes}
                            allowedSourceTypes={this.props.allowedSourceTypes}
                            value={this.props.sourceType}
                            onChange={this.handleChangeSourceType}
                        />

                        <InputGroup>
                            <InputLabel>Source Name</InputLabel>
                            <Input
                                value={this.props.sourceName}
                                onChange={this.handleChangeSourceName}
                                disabled={this.props.sourceType === "unknown"}
                            />
                        </InputGroup>
                    </IsolateFormFields>

                    <InputGroup>
                        <InputLabel>Isolate Name</InputLabel>
                        <Input value={formatIsolateName(this.props)} readOnly />
                    </InputGroup>
                </ModalBody>
                <ModalFooter>
                    <SaveButton />
                </ModalFooter>
            </form>
        );
    }
}

export default IsolateForm;
