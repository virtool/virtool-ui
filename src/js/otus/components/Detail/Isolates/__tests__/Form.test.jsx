import { shallow } from "enzyme";
import React from "react";
import { beforeEach, describe, expect, it, vi } from "vitest";
import { Input } from "../../../../../base";
import { SourceType } from "../../SourceType";
import { IsolateForm } from "../Form";

describe("<IsolateForm />", () => {
    let props;

    beforeEach(() => {
        props = {
            sourceType: "isolate",
            sourceName: "A",
            allowedSourceTypes: ["isolate", "genotype"],
            restrictSourceTypes: true,
            onChange: vi.fn(),
            onSubmit: vi.fn(),
        };
    });

    it("should render with source types restricted", () => {
        const wrapper = shallow(<IsolateForm {...props} />);
        expect(wrapper).toMatchSnapshot();
    });

    it("should render with source types unrestricted", () => {
        props.restrictSourceTypes = false;
        const wrapper = shallow(<IsolateForm {...props} />);
        expect(wrapper).toMatchSnapshot();
    });

    it.each([
        ["genotype", "genotype", "A"],
        ["unknown", "unknown", ""],
    ])("should call onChange() when source type changes to %p", (value, sourceType, sourceName) => {
        const e = {
            target: {
                value,
            },
        };
        const wrapper = shallow(<IsolateForm {...props} />);
        wrapper.find(SourceType).simulate("change", e);
        expect(props.onChange).toHaveBeenCalledWith({
            sourceName,
            sourceType,
        });
    });

    it("should call onChange() when source name changes", () => {
        const e = {
            target: {
                value: "B",
            },
        };
        const wrapper = shallow(<IsolateForm {...props} />);
        wrapper.find(Input).at(0).simulate("change", e);
        expect(props.onChange).toHaveBeenCalledWith({
            sourceName: "B",
            sourceType: "isolate",
        });
    });
});
