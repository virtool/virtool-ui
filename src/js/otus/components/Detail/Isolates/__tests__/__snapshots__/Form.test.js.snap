// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<IsolateForm /> should render with source types restricted 1`] = `
<form
  onSubmit={[MockFunction]}
>
  <Modal__ModalBody>
    <Form__IsolateFormFields>
      <SourceType
        allowedSourceTypes={
          Array [
            "isolate",
            "genotype",
          ]
        }
        onChange={[Function]}
        restrictSourceTypes={true}
        value="isolate"
      />
      <InputGroup>
        <Input__InputLabel>
          Source Name
        </Input__InputLabel>
        <Input
          disabled={false}
          onChange={[Function]}
          value="A"
        />
      </InputGroup>
    </Form__IsolateFormFields>
    <InputGroup>
      <Input__InputLabel>
        Isolate Name
      </Input__InputLabel>
      <Input
        readOnly={true}
        value="Isolate A"
      />
    </InputGroup>
  </Modal__ModalBody>
  <Modal__ModalFooter>
    <SaveButton />
  </Modal__ModalFooter>
</form>
`;

exports[`<IsolateForm /> should render with source types unrestricted 1`] = `
<form
  onSubmit={[MockFunction]}
>
  <Modal__ModalBody>
    <Form__IsolateFormFields>
      <SourceType
        allowedSourceTypes={
          Array [
            "isolate",
            "genotype",
          ]
        }
        onChange={[Function]}
        restrictSourceTypes={false}
        value="isolate"
      />
      <InputGroup>
        <Input__InputLabel>
          Source Name
        </Input__InputLabel>
        <Input
          disabled={false}
          onChange={[Function]}
          value="A"
        />
      </InputGroup>
    </Form__IsolateFormFields>
    <InputGroup>
      <Input__InputLabel>
        Isolate Name
      </Input__InputLabel>
      <Input
        readOnly={true}
        value="Isolate A"
      />
    </InputGroup>
  </Modal__ModalBody>
  <Modal__ModalFooter>
    <SaveButton />
  </Modal__ModalFooter>
</form>
`;
